### ISSUE TITLE
push_transaction can report error when transaction actually succeeds

### ISSUE THREAD

**Issue by heifner** [EOSIO/eos#4349]
chain_plugin push_transaction uses abi_serializer to create "pretty_output" json for the transaction_trace. If the `db.to_variant_with_abi` fails for any reason including the newly added serialization time limit of #4339 then the time limit assertion is reported to the user which from the user's perspective looks like the `push_transaction` failed when in actuality the transaction succeeded.

https://github.com/EOSIO/eos/blob/90fefdd12a4797e47890910f11dd7b60f1435ed4/plugins/chain_plugin/chain_plugin.cpp#L894

Error reporting should be improved to make it clear what exactly failed. In the case of the `push_transaction` the transaction_id should be reported in this case.

**Comment by tbfleming** [comment]
Tricky problem. Most users of `push_transaction` will assume this never occurs, because chances are high devs will never see it. When it does happen apps will break.

**Comment by tbfleming** [comment]
It's kind of like the surprise people get the first time they see just the transaction ID in a block. But even then, deferred transactions are more common than `to_variant_with_abi` failing.

**Comment by abourget** [comment]
I want to offer an endpoint that blocks  on `/v1/chain/push_transaction` until the transaction finds its way into a block.. it would watch the network, and return success (200) only if accepted.

To do that properly, I'd want to have a source of never-failing ABI (perhaps a node with high limit for abi serialization deadlines).. but I was also thinking of a special log entry, that would render the ABI to `greylog` and I'd be able to funnel through the nodes..

I'm also looking for a reliable data source for transaction results.. a data system can't tolerate having incomplete transaction data "just because" it's too long..

Would the abi serialization deadline still apply on an `edump()` call ?  Also, any way to ship that to a separate thread for serialization once transaction processing is done ?
